doc/model.rst

```diff
@@ -0,0 +1,932 @@
+When attaching an ML model using ``Model.add_ml_model()``, the
+following arguments are offered to customize the storage and execution of the ML model:
+
+- `name` (str): name to reference the model in the Orchestrator.
+- `backend` (str): name of the backend (TORCH, TF, TFLITE, ONNX).
+- `model` (t.Optional[str] = None): A model in memory (only supported for non-colocated orchestrators).
+- `model_path` (t.Optional[str] = None): serialized model.
+- `device` (t.Literal["CPU", "GPU"] = "CPU"): name of device for execution, defaults to “CPU”.
+- `devices_per_node` (int = 1): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- `first_device` (int = 0): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- `batch_size` (int = 0): batch size for execution, defaults to 0.
+- `min_batch_size` (int = 0): minimum batch size for model execution, defaults to 0.
+- `min_batch_timeout` (int = 0): time to wait for minimum batch size, defaults to 0.
+- `tag` (str = ""): additional tag for model information, defaults to “”.
+- `inputs` (t.Optional[t.List[str]] = None): model inputs (TF only), defaults to None.
+- `outputs` (t.Optional[t.List[str]] = None): model outputs (TF only), defaults to None.
+
+Continue for demonstrations on how to :ref:`load an in-memory ML model<in_mem_ML_model_ex>` and
+:ref:`load an ML model from file<from_file_ML_model_ex>`.
+
+.. _in_mem_ML_model_ex:
+-------------------------------------
+Example: Attach an in-memory ML Model
+-------------------------------------
+This example demonstrates how to attach an in-memory ML model to a SmartSim ``Model``
+to load into an ``Orchestrator`` at ``Model`` runtime.
+
+.. note::
+    This example assumes:
+
+    - an ``Orchestrator`` is launched prior to the ``Models`` execution
+    - an initialized ``Model`` named `smartsim_model` exists within the ``Experiment`` workflow
+
+**Define an in-memory a Keras CNN**
+
+The ML model must be defined using one of the supported ML frameworks. For the purpose of the example,
+we define a Keras CNN in the same script as the SmartSim ``Experiment``:
+
+.. code-block:: python
+
+    def create_tf_cnn():
+        """Create an in-memory Keras CNN for example purposes
+
+        """
+        from smartsim.ml.tf import serialize_model
+        n = Net()
+        input_shape = (3,3,1)
+        inputs = Input(input_shape)
+        outputs = n(inputs)
+        model = keras.Model(inputs=inputs, outputs=outputs, name=n.name)
+
+        return serialize_model(model)
+
+    # Get and save TF model
+    model, inputs, outputs = create_tf_cnn()
+
+**Attach the ML model to a SmartSim Model**
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add the in-memory TensorFlow model using
+the ``Model.add_ml_model()`` function and specify the model path to the parameter `model`:
```

Comment:
the ``Model.add_ml_model()`` function and specify the model path to the parameter `model`:

->

the ``Model.add_ml_model()`` function and specify the in-memory model to the parameter `model`: